src/lib/cosmos/components/Connectors.tsx

```diff
@@ -0,0 +1,230 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { formatEther } from '@ethersproject/units'
+import { SupportedChainId } from 'constants/chains'
+import { useAtom } from 'jotai'
+import { providerAtom } from 'lib/state'
+import styled from 'lib/theme'
+import { useEffect, useState } from 'react'
+import { Web3ReactHooks } from 'widgets-web3-react/core'
+import { MetaMask } from 'widgets-web3-react/metamask'
+import { Network } from 'widgets-web3-react/network'
+import { Connector } from 'widgets-web3-react/types'
+
+import { connectors } from '../connectors'
+import { hooks as metaMaskHooks, metaMask } from '../connectors/metaMask'
+import { hooks as networkHooks, network } from '../connectors/network'
+
+function getName(connector: Connector) {
+  if (connector instanceof MetaMask) {
+    return 'MetaMask'
+  } else if (connector instanceof Network) {
+    return 'Network'
+  } else {
+    return 'Unknown'
+  }
+}
+
+function Status({
+  connector,
+  hooks: { useChainId, useAccounts, useError },
+}: {
+  connector: Connector
+  hooks: Web3ReactHooks
+}) {
+  const chainId = useChainId()
+  const accounts = useAccounts()
+  const error = useError()
+
+  const connected = Boolean(chainId && accounts && accounts.length > 0)
```

Comment:
MetaMask continues to show as active after disconnection.

I think there is a bug in web3-react where the chainId persists despite disconnection. I also think there should be a hook that abstracts this "connected" logic - is that useIsActive()?

Reply:
Network does not require accounts to be connected.